### ISSUE TITLE
[BUG] nixvimInjections don't seem to work on nightly

### ISSUE THREAD

**Issue by jonboh** [nix-community/nixvim#1357]
| Field        | Description |
|--------------|-------------|
| Plugin       | `treesitter`  | 
| Nixpkgs      | `unstable` | 

- [x] I have read the [FAQ](https://nix-community.github.io/nixvim/user-guide/faq.html) and my bug is not listed there.

## Description

The nixvimInjections option does not work on nighlty builds of neovim. The extraConfigLua regions do not get properly highlighted as `lua` code. A minor thing, but I thought it worth to report :)

## Minimal, Reproducible Example (MRE)

I'm able to reproduce the behavior with this minimal flake:
```nix
{
  description = "A nixvim configuration";

  inputs = {
    nixvim.url = "github:nix-community/nixvim";
    flake-utils.url = "github:numtide/flake-utils";
    nixneovimplugins.url = "github:NixNeovim/NixNeovimPlugins";
    neovim-nightly-overlay.url = "github:nix-community/neovim-nightly-overlay";
  };

  outputs = {
    nixpkgs,
    nixvim,
    flake-utils,
    nixneovimplugins,
    ...
  } @ inputs:
    flake-utils.lib.eachDefaultSystem (system: let
      pkgs-nightly = import nixpkgs {
        inherit system;
        overlays = [
          nixneovimplugins.overlays.default
          inputs.neovim-nightly-overlay.overlay
        ];
      };
      pkgs = import nixpkgs {
        inherit system;
        overlays = [
          nixneovimplugins.overlays.default
        ];
      };
      nixvim' = nixvim.legacyPackages.${system};
      inject = nixvim'.makeNixvimWithModule {
        inherit pkgs;
        module = {
          plugins.treesitter = {
            enable = true;
            nixvimInjections = true;
          };
        };
      };
      inject-nightly = nixvim'.makeNixvimWithModule {
        pkgs = pkgs-nightly;
        module = {
          plugins.treesitter = {
            enable = true;
            nixvimInjections = true;
          };
        };
      };
    in {
      packages = {
        inherit inject;
        inherit inject-nightly;
      };
    });
}
```
`nix run ".#inject"` and `nix run ".#inject-nightly"` for the different versions.
Opening any file with `extraConfigLua` should show that the nighly version does not properly highlight those regions.

**Comment by elohmeier** [comment]
Can confirm this for [current nixpkgs-unstable](https://github.com/NixOS/nixpkgs/commit/d226935fd75012939397c83f6c385e4d6d832288) with Neovim v0.10.0.

**Comment by traxys** [comment]
Should be fixed in #1843

**Comment by zvolin** [comment]
This still doesn't work for me on latest nixvim and nixpkgs.

**Comment by MattSturgeon** [comment]
@zvolin you may be interested in this discussion: https://github.com/nix-community/nixvim/discussions/2093

TL;DR: the injections are working, but can be a little finicky when it comes to whitespace.

Also: ensure you have set `plugins.treesitter.settings.highlight.enable = true`.

**Comment by zvolin** [comment]
thanks, `highlight.enable = true` did the job. I think it was working without it few months ago.

**Comment by khaneliman** [comment]
> thanks, `highlight.enable = true` did the job. I think it was working without it few months ago.

We used to default to enabling them but changed to respecting plugin defaults as we’ve been refactoring. Treesitter doesn’t enable anything by default apparently.

**Comment by dzervas** [comment]
I have the same issue. I also tried adding custom injections in `after/queries/yaml/injections.scm` and `queries/yaml/injections.scm` both through extraFiles and manually but nothing worked - treesitter doesn't pick any of the files up, like they don't exist
